--- json-c-0.9/json_object.c    2011-08-12 15:53:35.000000000 +0200
+++ json-c-0.9/json_object.c    2011-08-12 15:49:44.000000000 +0200
@@ -97,6 +97,7 @@
     case '"':
     case '\\':
     case '/':
+    case '%':
       if(pos - start_offset > 0)
        printbuf_memappend(pb, str + start_offset, pos - start_offset);
       if(c == '\b') printbuf_memappend(pb, "\\b", 2);
@@ -106,8 +107,10 @@
       else if(c == '"') printbuf_memappend(pb, "\\\"", 2);
       else if(c == '\\') printbuf_memappend(pb, "\\\\", 2);
       else if(c == '/') printbuf_memappend(pb, "\\/", 2);
+      else if(c == '%') printbuf_memappend(pb, "%%", 2);
       start_offset = ++pos;
       break;
+
     default:
       if(c < ' ') {
        if(pos - start_offset > 0)

